packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -0,0 +1,297 @@
+        return new DidCommService({
+          id: `${connectionRecord?.did};#${s.priority}`,
+          priority: 0,
+          serviceEndpoint: s.serviceEndpoint,
+          recipientKeys: s.recipientKeys,
+          routingKeys: s.routingKeys,
+        })
+      })
+
+      // Discover what handshake protocols are supported
+      const handshakeProtocols = this.getSupportedHandshakeProtocols()
+
+      const options = {
+        ...config,
+        accept: ['didcomm/aip1'],
+        services,
+        handshakeProtocols,
+      }
+      outOfBandMessage = new OutOfBandMessage(options)
+    } else {
+      const routing = await this.mediationRecipientService.getRouting()
+      const service = new DidCommService({
+        id: '#inline',
+        priority: 0,
+        serviceEndpoint: routing.endpoints[0],
+        recipientKeys: [routing.verkey],
+        routingKeys: routing.routingKeys,
+      })
+      const options = {
+        ...config,
+        accept: ['didcomm/aip1'],
+        services: [service],
+      }
+      outOfBandMessage = new OutOfBandMessage(options)
+    }
+
+    if (messages) {
+      messages.forEach((message) => {
+        if (message.service) {
+          // We can remove `~service` attribute from message. Newer OOB messages have `services` attribute instead.
+          message.service = undefined
+        }
+        outOfBandMessage.addRequest(message)
+      })
+    }
+
+    return { outOfBandMessage, connectionRecord }
+  }
+
+  public async receiveInvitationFromUrl(urlMessage: string, config: ReceiveOutOfBandMessageConfig) {
+    const parsedUrl = parseUrl(urlMessage).query
+    if (parsedUrl['oob']) {
+      const outOfBandMessage = await OutOfBandMessage.fromUrl(urlMessage)
+      return this.receiveMessage(outOfBandMessage, config)
+    } else if (parsedUrl['c_i'] || parsedUrl['d_m']) {
+      const invitation = await ConnectionInvitationMessage.fromUrl(urlMessage)
+      return this.connectionsModule.receiveInvitation(invitation)
+    }
+  }
+
+  /**
+   * Takes all messages from `requests~attach` attribute and pass them to the agent via event emitter.
+   *
+   * @param outOfBandMessage
+   */
+  public async receiveMessage(
+    outOfBandMessage: OutOfBandMessage,
+    config: ReceiveOutOfBandMessageConfig
+  ): Promise<ConnectionRecord | undefined> {
+    const { handshakeProtocols, services } = outOfBandMessage
+    const { autoAccept, reuse } = config
+    const messages = outOfBandMessage.getRequests()
+
+    if ((!handshakeProtocols || handshakeProtocols.length === 0) && (!messages || messages?.length === 0)) {
+      throw new AriesFrameworkError(
+        'One or both of handshake_protocols and requests~attach MUST be included in the message.'
+      )
+    }
+
+    if (handshakeProtocols) {
+      if (!this.areHandshakeProtocolsSupported(handshakeProtocols)) {
+        throw new AriesFrameworkError('Handshake protocols are not supported.')
```

Comment:
Maybe would be good to include in the error which handshakeprotocols were requestsed vs which are supported